README.md

```diff
@@ -4,17 +4,270 @@
 
-* [Usage](#usage)
-   * [Tracing nginx](#tracing-nginx)
-      * [Quick start/Example](#quick-start-with-docker-example)
-      * [Guide](#guide)
-* [Development](#building)
+- [Datadog OpenTracing C++ Client](#datadog-opentracing-c-client)
+  - [Usage](#usage)
+    - [Tracing C++ Applications](#tracing-c-applications)
+      - [Getting Started](#getting-started)
+      - [Compatibility](#compatibility)
+      - [Installation](#installation)
+        - [Compile against dd-opentracing-cpp](#compile-against-dd-opentracing-cpp)
+        - [Dynamic Loading](#dynamic-loading)
+      - [Advanced Usage](#advanced-usage)
+        - [OpenTracing](#opentracing)
+        - [Manual Instrumentation](#manual-instrumentation)
+        - [Custom Tagging](#custom-tagging)
+        - [Distributed Tracing](#distributed-tracing)
+        - [Priority Sampling](#priority-sampling)
+        - [Logging](#logging)
+        - [Debugging](#debugging)
+    - [Tracing Nginx](#tracing-nginx)
+      - [Quick-start with Docker example](#quick-start-with-docker-example)
+      - [Guide](#guide)
+    - [Tracing Envoy & Istio](#tracing-envoy--istio)
+  - [Building](#building)
 
 ## Usage
 
-### Tracing C++ Code
+### Tracing C++ Applications
 
-Support coming soon.
+#### Getting Started
+
+To begin tracing applications written in any language, first [install and configure the Datadog Agent](https://docs.datadoghq.com/tracing/setup).
+
+You will need to compile against [OpenTracing-cpp](https://github.com/opentracing/opentracing-cpp).
+
+#### Compatibility
+
+dd-opentracing-cpp needs C++14 to build, but if you use [dynamic loading](#dynamic-loading) then you are instead limited by OpenTracing's requirement for [C++11 or later](https://github.com/opentracing/opentracing-cpp/#cc98).
+
+Supported platforms are: Linux & Mac. If you need Windows support, please let us know.
+
+#### Installation
+
+Datadog tracing can be enabled in one of two ways:
+
+* Compile against dd-opentracing-cpp, where the Datadog lib is compiled in and configured in code
+* Dynamic loading, where the Datadog OpenTracing library is loaded at run-time and configured via JSON
+
+##### Compile against dd-opentracing-cpp
+
+```bash
+# Download and install dd-opentracing-cpp library.
+wget https://github.com/DataDog/dd-opentracing-cpp/archive/v0.3.5.tar.gz -O dd-opentracing-cpp.tar.gz
+tar zxvf dd-opentracing-cpp.tar.gz
+mkdir dd-opentracing-cpp-0.3.5/.build
+cd dd-opentracing-cpp-0.3.5/.build
+# Download and install the correct version of opentracing-cpp, & other deps.
+../scripts/install_dependencies.sh
+cmake ..
+make
+make install
+```
+
+Include `<datadog/opentracing.h>` and create the tracer:
+
+```cpp
+// tracer_example.cpp
+#include <datadog/opentracing.h>
+#include <iostream>
+#include <string>
+
+int main(int argc, char* argv[]) {
+  datadog::opentracing::TracerOptions tracer_options{"dd-agent", 8126, "compiled-in example"};
```

Comment:
This should probably be `localhost`

Reply:
I made it the same code as what is used in the example, which runs on Docker.